pagopa/pn-frontend
# fix/PN-6104 - PG - notification detail - show recipient denomination for multirecipient notifications in delegate access

## Short description
Fixed a bug: in PG, when the user enters the detail of a multirecipient notification as a delegate, the denomination in the upper-left box didn't include the denomination of the recipient which delegates to the user.  
To this end, we decided to trust the API response info about recipients: only the details for the delegate recipient should be complete, for the other recipients only the recipientType is expected. This differs from the previously drafted solution of searching in the set of all delegators of the logged user/company.  
The solution adopted implies a medium-size refactoring, mainly to clean code which has no sense given the implemented approach.

## List of changes proposed in this pull request
- Change the algorithm to decide the currentRecipient in the function which parses the API response for the details of a notification. 
- Modified the signature of this function since some arguments are no longer needed (given the implemented approach), modified other signature (e.g. of the API call) accordingly.
- Adapted several tests to align them with the modified signatures.

## How to test
Enter PG app with a user/company which is delegate for other entities, go to "Notifiche delegate" option, enter the detail for a multirecipient notification. The denomination of the delegator should appear in the upper-left box in the notification detail.  
Example: user DanteAlighieri, company DivinaCommedia srl, two notfications for Ada Lovelace (delegator of DanteAlighieri/DivinaCommedia srl) and another recipient
  - IUN DJPX-VHUP-MRVK-202305-D-1 (Ada Lovelace is first recipient)
  - IUN YMJW-JXTN-HLUQ-202305-Q-1 (Ada Lovelace is second recipient)
